scipy/_lib/deprecation.py

```diff
@@ -0,0 +1,17 @@
+import functools
+import warnings
+
+__all__ = ["_deprecated"]
+
+
+def _deprecated(msg):
+    """Deprecate a function by emitting a warning on use."""
+    def wrap(fun):
+        @functools.wraps(fun)
+        def call(*args, **kwargs):
+            warnings.warn(msg, category=DeprecationWarning)
```

Comment:
This probably needs `stacklevel=2`?

Reply:
You were correct with `stacklevel=2` originally, test added